src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -494,19 +496,19 @@ protected FieldValueInfo completeValueForList(ExecutionContext executionContext,
      */
     protected FieldValueInfo completeValueForList(ExecutionContext executionContext, ExecutionStrategyParameters parameters, Iterable<Object> iterableValues) {
 
-        Collection<Object> values = FpKit.toCollection(iterableValues);
+        OptionalInt size = FpKit.toSize(iterableValues);
         ExecutionStepInfo executionStepInfo = parameters.getExecutionStepInfo();
 
-        InstrumentationFieldCompleteParameters instrumentationParams = new InstrumentationFieldCompleteParameters(executionContext, parameters, () -> executionStepInfo, values);
+        InstrumentationFieldCompleteParameters instrumentationParams = new InstrumentationFieldCompleteParameters(executionContext, parameters, () -> executionStepInfo, iterableValues);
```

Comment:
This is a mild breaking change - the input is "Object" but the actual type has changed.

I think this is ok (did we say it WOULD always be a collection??) but I just want to call it out and see others thoughts on this.

Reply:
yes, this could be a breaking change.
`Arrays` are reallocated anymore as `List `(but I could easily re-introduce this behavior).

For Collections we should be fine: `toIterable` does not change the concrete type.
For Streams/Iterators we should be fine as well: nobody is using that right now.